### PR TITLE
Doors and ports

### PR DESCRIPTION
**Pull request by JaimieGR**
Adds doors and trapdoors as well as stairs just cause

### CODE REVIEW

**Comment by JaimieGR** [comment]
@macmv doors are not dropping themselves nor does middle click work on them they currently drop oak doors could use some help with that

File: src/main/java/net/macmv/rgen/block/RgenTrapDoors.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+public class RgenTrapDoors extends BlockTrapDoor {
+
+  public RgenTrapDoors() {
+    super(Material.WOOD);
+    this.setHardness(3.0F);
+    this.setResistance(5.0F);
+    this.setSoundType(SoundType.WOOD);
+  }
+
+}
```
Comment (macmv) [inline src/main/java/net/macmv/rgen/block/RgenTrapDoors.java:16]: this organization is great, do this more.

File: src/main/java/net/macmv/rgen/block/BambooDoor.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package net.macmv.rgen.block;
+
+public class BambooDoor extends RgenDoor {
+  // You can add custom properties or methods for the BambooDoor here if needed
```
Comment (macmv) [inline src/main/java/net/macmv/rgen/block/BambooDoor.java]: don't organize like this though, its confusing and empty classes are pointless

File: src/main/java/net/macmv/rgen/block/RgenTrapDoors.java (hunk 1 above)
Comment (macmv) [inline src/main/java/net/macmv/rgen/block/RgenTrapDoors.java:16]: this organization is great, do this more.

File: src/main/java/net/macmv/rgen/item/RItems.java [hunk 3]
```diff
@@ -49,6 +51,31 @@ public class RItems {
...
+  public static final Item FIR_STAIRS = registerItem("fir_stairs", new ItemBlock(RBlocks.FIR_STAIRS));
+  public static final Item PALM_STAIRS = registerItem("palm_stairs", new ItemBlock(RBlocks.PALM_STAIRS));
+
+  public static final Item DEAD_TRAPDOOR = registerItem("dead_trapdoor", new ItemBlock(RBlocks.DEAD_TRAPDOOR));
+  public static final Item MANGROVE_TRAPDOOR = registerItem("mangrove_trapdoor", new ItemBlock(RBlocks.MANGROVE_TRAPDOOR));
+  //public static final Item BAMBOO_TRAPDOOR = registerItem("bamboo_trapdoor", new ItemBlock(RBlocks.BAMBOO_TRAPDOOR));
+  public static final Item CEDAR_TRAPDOOR = registerItem("cedar_trapdoor", new ItemBlock(RBlocks.CEDAR_TRAPDOOR));
+  public static final Item CHERRY_TRAPDOOR = registerItem("cherry_trapdoor", new ItemBlock(RBlocks.CHERRY_TRAPDOOR));
+  public static final Item FIR_TRAPDOOR = registerItem("fir_trapdoor", new ItemBlock(RBlocks.FIR_TRAPDOOR));
+  public static final Item PALM_TRAPDOOR = registerItem("palm_trapdoor", new ItemBlock(RBlocks.PALM_TRAPDOOR));
```
Comment (macmv) [inline src/main/java/net/macmv/rgen/item/RItems.java]: stairs and trapdoors should be registered with `registerBlockItem`, not `registerItem`.

File: src/main/resources/assets/rgen/models/block/stairs/vanilla/inner_stairs.bbmodel
Comment (macmv) [inline src/main/resources/assets/rgen/models/block/stairs/vanilla/inner_stairs.bbmodel:1]: can you delete this if its unused?

File: src/main/resources/assets/rgen/models/block/stairs/vanilla/outer_stairs.bbmodel
Comment (macmv) [inline src/main/resources/assets/rgen/models/block/stairs/vanilla/outer_stairs.bbmodel:1]: can you delete this if its unused?

**Review by macmv — COMMENTED** [review]
looks good, just a couple nits

File: src/main/java/net/macmv/rgen/block/BambooDoor.java (hunk 2 above)
Comment (JaimieGR) [reply src/main/java/net/macmv/rgen/block/BambooDoor.java]: 485b8d30ab4a547b61caefdb359946de5f08bb33 resolved

File: src/main/java/net/macmv/rgen/block/RgenTrapDoors.java (hunk 1 above)
Comment (JaimieGR) [reply src/main/java/net/macmv/rgen/block/RgenTrapDoors.java:16]: 485b8d30ab4a547b61caefdb359946de5f08bb33 resolved

File: src/main/java/net/macmv/rgen/item/RItems.java (hunk 3 above)
Comment (JaimieGR) [reply src/main/java/net/macmv/rgen/item/RItems.java]: 485b8d30ab4a547b61caefdb359946de5f08bb33 resolved

File: src/main/resources/assets/rgen/models/block/stairs/vanilla/inner_stairs.bbmodel
Comment (JaimieGR) [reply src/main/resources/assets/rgen/models/block/stairs/vanilla/inner_stairs.bbmodel:1]: 485b8d30ab4a547b61caefdb359946de5f08bb33 resolved

File: src/main/resources/assets/rgen/models/block/stairs/vanilla/outer_stairs.bbmodel
Comment (JaimieGR) [reply src/main/resources/assets/rgen/models/block/stairs/vanilla/outer_stairs.bbmodel:1]: 485b8d30ab4a547b61caefdb359946de5f08bb33 resolved

**Pull request merged**